internal/static/static.go

```diff
@@ -0,0 +1,59 @@
+package static
+
+import (
+	"go/types"
+
+	"github.com/dogmatiq/configkit"
+	"golang.org/x/tools/go/packages"
+	"golang.org/x/tools/go/ssa"
+	"golang.org/x/tools/go/ssa/ssautil"
+)
+
+const (
+	// dogmaPkgPath is the full path of dogma package.
+	dogmaPkgPath = "github.com/dogmatiq/dogma"
+)
+
+// FromPackages loads the list of configkit.Application from the given list
+// of packages.Package items.
+func FromPackages(pkgs []*packages.Package) []configkit.Application {
+	prog, packages := ssautil.AllPackages(pkgs, ssa.SanityCheckFunctions)
+	prog.Build()
+
+	dogmaPkg := prog.ImportedPackage(dogmaPkgPath)
+	if dogmaPkg == nil {
+		// If Dogma package is not found as an import, none of the packages in
+		// the built application implement dogma.Application interface.
+		return nil
+	}
+
+	var apps []configkit.Application
+	iface := dogmaPkg.Pkg.Scope().Lookup("Application").Type().Underlying().(*types.Interface)
+
+	for _, pkg := range packages {
+		if pkg == nil {
+			continue
```

Comment:
```suggestion
			// Any packages.Package that can not be built results in a nil
			// ssa.Package. We ignore any such packages so that we can still
			// obtain information about applications from any valid packages.
			continue
```